var/spack/repos/builtin/packages/vtk/package.py

```diff
@@ -51,7 +51,6 @@ class Vtk(CMakePackage):
 
     # VTK7 defaults to OpenGL2 rendering backend
     variant("opengl2", default=True, description="Enable OpenGL2 backend")
-    variant("osmesa", default=False, description="Enable OSMesa support")
```

Comment:
So looking at this again, we want to be able to specify this variant for the case we build X and OSMesa versions in the same environment. If VTK is a root spec, that is no longer possible. Ref. ParaView and VisIt

Reply:
Not sure I understand this use case. Modulo the fact that the `depends_on` directives on develop are currently wrong, I assume the intended behavior for `+osmesa` was to select `osmesa` as a `gl` provider. If my understanding is correct, then the variant is not necessary, and one can simply say:
```
vtk ^[virtuals=gl] osmesa
```